Java/src/main/java/net/hypixel/api/util/PropertyFilter.java

```diff
@@ -0,0 +1,309 @@
+     * are considered to be colliding. Some notes about key collision:
+     * <pre>
+     *     • New keys always take precedence when collision happens, such that...
+     *     • If an added key has a <u>wider scope</u> (fewer tokens) than any existing keys, all
+     *       colliding keys with narrower scopes (more tokens) will be removed, and the new key will
+     *       be added.
+     *     • If the added key has a <u>narrower scope</u> (more tokens) than an existing key, the
+     *       colliding key with a wider scope (fewer tokens) will be replaced with the new one.
+     * </pre>
+     * An example of collision would be adding {@code "stats.SkyWars.coins"} when {@code
+     * "stats.SkyWars"} is already included. In that case, the added key would replace the existing
+     * key, so that only {@code "stats.SkyWars.coins"} is included when filtering.
+     *
+     * @param keys Names of properties that will be allowed to pass through the filter (in
+     *             dot-notation).
+     */
+    public void include(String... keys) {
+        if (keys == null) {
+            throw new IllegalArgumentException("Cannot include null property keys");
+        }
+
+        // Check for key collisions.
+        for (String rawKey : keys) {
+            if (rawKey == null) {
+                throw new IllegalArgumentException("Cannot include null property keys");
+            }
+
+            PropertyKey key = new PropertyKey(rawKey);
+            boolean shouldAddKey = true;
+            Iterator<PropertyKey> existingKeys = allowedKeys.iterator();
+
+            while (existingKeys.hasNext()) {
+                PropertyKey existingKey = existingKeys.next();
+
+                // Ignore duplicate keys.
+                if (existingKey.equals(key)) {
+                    shouldAddKey = false;
+                    break;
+                }
+
+                // Check if the new key collides with the existing key's scope.
+                if (key.isExtendedBy(existingKey)) {
+                    // Replace & continue, since there can be multiple keys with narrower scopes.
+                    existingKeys.remove();
+                } else if (existingKey.isExtendedBy(key)) {
+                    // Replace & break, since only 1 key should possibly have a wider scope.
+                    existingKeys.remove();
+                    break;
+                }
+            }
+
+            if (shouldAddKey) {
+                allowedKeys.add(key);
+            }
+        }
+    }
+
+    /**
+     * Removes all of the provided keys from the filter, such that objects {@link
+     * #applyTo(ComplexHypixelObject) passed through} the filter will <em>not</em> include
+     * properties with those keys.
+     * <p><br>
+     * Attempting to remove a key that was already removed, or never {@link #include(String...)
+     * included} to begin with, will have no effect.
+     */
+    public void removeKeys(String... keys) {
```

Comment:
Either `includeKeys` and `removeKeys` or `include` and `remove`

Reply:
Alright, sticking to the latter